.github/workflows/multi-approvers.js

```diff
@@ -2,49 +2,120 @@ const APPROVED = 'APPROVED';
 const COMMENTED = 'COMMENTED';
 const MIN_APPROVED_COUNT = 2;
 
-/** Returns the number of approvals from members in the given list. */
-function inOrgApprovedCount(members, submittedReviews, prLogin) {
-  const reviewStateByLogin = {};
-  submittedReviews
-    // Remove the PR user.
-    .filter((r) => r.user.login !== prLogin)
-    // Only consider users in the org.
-    .filter((r) => members.has(r.user.login))
-    // Sort chronologically ascending. Note that a reviewer can submit multiple reviews.
-    .sort((a, b) => new Date(a.submitted_at) - new Date(b.submitted_at))
-    .forEach((r) => {
-      const reviewerLogin = r.user.login;
-
-      // Set state if it does not exist.
-      if (!Object.hasOwn(reviewStateByLogin, reviewerLogin)) {
-        reviewStateByLogin[reviewerLogin] = r.state;
-        return;
-      }
+/** Implements approvedCount using the contains method. */
+class AbstractMembers {
+
+  /** Returns the number of approvals from members in the given list. */
+  async approvedCount(submittedReviews, login) {
+    const reviewStateByLogin = {};
+    submittedReviews
+      // Remove the PR user.
+      .filter((r) => r.user.login !== prLogin)
+      // Only consider users in the org.
+      .filter((r) => this.contains(r.user.login))
+      // Sort chronologically ascending. Note that a reviewer can submit multiple reviews.
+      .sort((a, b) => new Date(a.submitted_at) - new Date(b.submitted_at))
+      .forEach((r) => {
+        const reviewerLogin = r.user.login;
+
+        // Set state if it does not exist.
+        if (!Object.hasOwn(reviewStateByLogin, reviewerLogin)) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+          return;
+        }
+
+        // Always update state if not approved.
+        if (reviewStateByLogin[reviewerLogin] !== APPROVED) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+          return;
+        }
+
+        // Do not update approved state for a comment.
+        if (reviewStateByLogin[reviewerLogin] === APPROVED && r.state !== COMMENTED) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+        }
+      })
+
+    return Object.values(reviewStateByLogin).filter((s) => s === APPROVED).length;
```

Comment:
This is super weird and would be a lot clearer with a single `for` loop using the quick `continue` pattern.

I also don't really understand why we're putting things into a map, then returning the value? It seems like a lot of unnecessary filtering and iteration.

Reply:
Please create a follow-up issue.